### DISCUSSION TITLE
Username is different from the "preferred_username" of OpenID

### DISCUSSION THREAD

**Discussion by loelkes** [semaphoreui/semaphore#1745] [Q&A]
Context: Semaphore v2.9.37 running in Docker, with OpenID (from Keycloak). Everything works as expected with one exception. 

In `config.json` I've set

```
...
    "username_claim": "preferred_username",
...
```

Instead of `loelkes` (my `preferred_username`) my username has (what it seems) a random string attached to it.

<img width="427" alt="Screenshot 2023-12-22 at 19 26 58" src="https://github.com/ansible-semaphore/semaphore/assets/231909/14e89d65-a155-410f-b32d-a86acd289a3a">

Is this intentional, and if so, why?

**Comment by fiftin** [comment]
Hi, it is not a bug. It is a way to work around the username conflict when you use several OpenID providers.
For example you use GitHub and GitLab auths. You have username `user1` on GitHub and `user2` on GitLab but same email.

**Reply by loelkes** [reply]
Why is it a problem if two users have the same email if `username_claim` is set to `preferred_username`?

My expectation here would be that if I have IdP 1 with `myname` / `myname@domain` and IdP 2 with `myname` / `myname@domain` that both link to the same user `myname`.
